### PR TITLE
Enables support for batched evaluations.

### PR DESCRIPTION
**Pull request by armfazh**
Based on top of #13

### CODE REVIEW

File: src/oprf.ts [hunk 1]
```diff
@@ -137,83 +144,68 @@ export class Evaluation {
...
-    }
-
     static deserialize(params: DLEQParams, bytes: Uint8Array): Evaluation {
-        checkSize(bytes, Evaluation, params)
-        const eltSize = Elt.size(params.gg)
-        const evaluated = Elt.deserialize(params.gg, bytes.subarray(0, eltSize))
+        const { head: evalList, tail } = fromU16LenPrefixClass(Elt, params.gg, bytes)
         let proof: DLEQProof | undefined
-        if (bytes[eltSize as number] === 1) {
+        if (tail[0] === 1) {
```
Comment (lukevalenta) [inline src/oprf.ts:157]: Making this `1` into a constant might make it more clear that it's an indicator of whether or not the proof is included (see other comment).

File: src/oprf.ts [hunk 2]
```diff
@@ -137,83 +144,68 @@ export class Evaluation {
...
 
-    static size(params: DLEQParams): number {
-        return Elt.size(params.gg) + 1
-    }
-
     static deserialize(params: DLEQParams, bytes: Uint8Array): Evaluation {
-        checkSize(bytes, Evaluation, params)
-        const eltSize = Elt.size(params.gg)
-        const evaluated = Elt.deserialize(params.gg, bytes.subarray(0, eltSize))
+        const { head: evalList, tail } = fromU16LenPrefixClass(Elt, params.gg, bytes)
```
Comment (lukevalenta) [inline src/oprf.ts:155]: Renaming `tail` to something like `optionalProof` could improve clarity here.

**Review by lukevalenta — APPROVED** [review]
A couple minor comments, but look good overall!

File: src/oprf.ts [hunk 3]
```diff
@@ -123,11 +130,11 @@ export abstract class Oprf {
...
 
 export class Evaluation {
-    constructor(public readonly evaluated: Elt, public readonly proof?: DLEQProof) {}
+    constructor(public readonly evaluated: Array<Elt>, public readonly proof?: DLEQProof) {}
 
     serialize(): Uint8Array {
         return joinAll([
-            this.evaluated.serialize(true),
+            ...toU16LenPrefixClass(this.evaluated),
             Uint8Array.from([this.proof ? 1 : 0]),
```
[the remark this answers is not in the document]
Comment (armfazh) [reply src/oprf.ts:138]: you pointed me to a better idea: serialize the OPRF mode, so we know whether a proof exists or not depending on the mode.

**Pull request merged**